"""
This is doctest for lazy_matrix_mul function
>>> with open("101-lazy_matrix_mul.py", "r") as f:
...     code = f.read()
>>> exec(code)
>>> m_a = [[1, 2], [3, 4]]
>>> m_b = [[5, 6], [7, 8]]
>>> lazy_matrix_mul(m_a, m_b)
array([[19, 22],
       [43, 50]])
>>> m_a = [[1, 2], [3, 4], [3, 4]]
>>> m_b = [[5, 6, 1], [7, 8, 2]]
>>> lazy_matrix_mul(m_a, m_b)
array([[19, 22,  5],
       [43, 50, 11],
       [43, 50, 11]])
>>> m_a = "Holberton"
>>> m_b = [[5, 6], [7, 8]]
>>> lazy_matrix_mul(m_a, m_b)
Traceback (most recent call last):
    ...
TypeError: Scalar operands are not allowed, use '*' instead
>>> m_a = [[5, 6], [7, 8]]
>>> m_b = "Holberton"
>>> lazy_matrix_mul(m_a, m_b)
Traceback (most recent call last):
    ...
TypeError: Scalar operands are not allowed, use '*' instead
>>> m_a = [1, 2, 3, 4]
>>> m_b = [[5, 6], [7, 8]]
>>> lazy_matrix_mul(m_a, m_b)
Traceback (most recent call last):
    ...
TypeError: m_a must be a list of lists
>>> m_a = [[5, 6], [7, 8]]
>>> m_b = [1, 2, 3, 4]
>>> lazy_matrix_mul(m_a, m_b)
Traceback (most recent call last):
    ...
TypeError: m_b must be a list of lists
>>> m_a = [[]]
>>> m_b = [[5, 6], [7, 8]]
>>> lazy_matrix_mul(m_a, m_b)
Traceback (most recent call last):
    ...
ValueError: shapes (1, 0) and (2, 2) not aligned: 0 (dim 1) != 2 (dim 0)
>>> m_a = [[5, 6], [7, 8]]
>>> m_b = [[]]
>>> lazy_matrix_mul(m_a, m_b)
Traceback (most recent call last):
    ...
ValueError: shapes (1, 0) and (2, 2) not aligned: 0 (dim 1) != 2 (dim 0)
>>> m_a = [[5, "6"], [7, 8]]
>>> m_b = [[5, 6], [7, 8]]
>>> lazy_matrix_mul(m_a, m_b)
Traceback (most recent call last):
    ...
TypeError: invalid data type for einsum
>>> m_a = [[5, 6], [7, 8]]
>>> m_b = [[5, "6"], [7, 8]]
>>> lazy_matrix_mul(m_a, m_b)
Traceback (most recent call last):
    ...
TypeError: invalid data type for einsum
>>> m_a = [[5, 6, 10], [7, 8]]
>>> m_b = [[5, 6], [7, 8]]
>>> lazy_matrix_mul(m_a, m_b)
Traceback (most recent call last):
    ...
TypeError: setting an array element with a sequence.
>>> m_a = [[5, 6], [7, 8]]
>>> m_b = [[5, 6, 1], [7, 8]]
>>> lazy_matrix_mul(m_a, m_b)
Traceback (most recent call last):
    ...
TypeError: setting an array element with a sequence.
>>> m_a = [[1, 2, 3], [3, 4, 5]]
>>> m_b = [[1, 2], [3, 4]]
>>> lazy_matrix_mul(m_a, m_b)
Traceback (most recent call last):
    ...
ValueError: shapes (2, 3) and (2, 2) not aligned: 3 (dim 1) != 2 (dim 0)
>>> lazy_matrix_mul()
Traceback (most recent call last):
    ...
TypeError: lazy_matrix_mul() missing 2 required positional arguments: 'm_a' and 'm_b'
>>> m_a = [[1, 2, 3], [3, 4, 5]]
>>> lazy_matrix_mul(m_a)
Traceback (most recent call last):
    ...
TypeError: lazy_matrix_mul() missing 1 required positional argument: 'm_b'

"""
